earn/src/components/borrow/modal/RemoveCollateralModal.tsx

```diff
@@ -64,25 +71,34 @@ function RemoveCollateralButton(props: RemoveCollateralButtonProps) {
   const { activeChain } = useContext(ChainContext);
 
   const [isPending, setIsPending] = useState(false);
+  const [oracleSeed, setOracleSeed] = useChainDependentState<number | undefined>(undefined, activeChain.id);
+
+  const provider = useProvider({ chainId: activeChain.id });
 
   const isToken0Collateral = collateralToken.address === marginAccount.token0.address;
 
   const amount0 = isToken0Collateral ? collateralAmount : GN.zero(collateralToken.decimals);
   const amount1 = isToken0Collateral ? GN.zero(collateralToken.decimals) : collateralAmount;
 
+  useEffectOnce(() => {
+    (async () => {
+      const seed = await computeOracleSeed(marginAccount.uniswapPool, provider, activeChain.id);
+      setOracleSeed(seed);
+    })();
+  });
+
+  const borrowerInterface = useMemo(() => new ethers.utils.Interface(borrowerABI), []);
+  const encodedData = useMemo(
+    () => borrowerInterface.encodeFunctionData('transfer', [amount0.toBigNumber(), amount1.toBigNumber(), userAddress]),
+    [amount0, amount1, borrowerInterface, userAddress]
+  );
```

Comment:
Probably don't need useMemo here at all, but if you want to use it, at least move the `borrowerInterface` declaration inside the second `useMemo`. No reason to have them separate.